caliban_toolbox/build.py

```diff
@@ -0,0 +1,138 @@
+#
+# Licensed under a modified Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.github.com/vanvalenlab/caliban-toolbox/LICENSE
+#
+# The Work provided may be used for non-commercial academic purposes only.
+# For any other use of the Work, including commercial use, please contact:
+# vanvalenlab@gmail.com
+#
+# Neither the name of Caltech nor the names of its contributors may be used
+# to endorse or promote products derived from this software without specific
+# prior written permission.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import math
+
+import numpy as np
+
+from deepcell_toolbox.utils import resize, tile_image
+
+
+def pad_image_stack(images, crop_size):
+    """Pads an an array of images so that it is divisible by the specified crop_size
+
+    Args:
+        images: array of images to be cropped
+        crop_size: tuple specifying crop size
+
+    Returns:
+        np.array: padded image stack
+    """
+
+    row_len, col_len = images.shape[1:3]
+    row_crop, col_crop = crop_size
+    row_num = math.ceil(row_len / crop_size[0])
+    col_num = math.ceil(col_len / crop_size[1])
+
+    new_row_len = row_num * row_crop
+    new_col_len = col_num * col_crop
+
+    if new_row_len == row_len and new_col_len == col_len:
+        # don't need to pad
+        return images
+    else:
+        new_images = np.zeros((images.shape[0], new_row_len, new_col_len, images.shape[3]))
+        new_images[:, :row_len, :col_len, :] = images
+        return new_images
+
+
+def pad_image_edge_crops(images, crop_size):
+    """Add padding to an image to enable additional edge crops to be generated
+
+    Args:
+        images: array of images to crop
+        crop_size: size of crops that will be produced
+
+    Returns:
+        np.array: padded image
+    """
+    row_offset, col_offset = int(crop_size[0] / 2), int(crop_size[1] / 2)
+
+    padded_images = np.zeros((images.shape[0], images.shape[1] + crop_size[0],
+                              images.shape[2] + crop_size[1], images.shape[3]))
+
+    padded_images[:, row_offset:-row_offset, col_offset:-col_offset, :] = images
+
+    return padded_images
+
+
+def combine_npz_files(npz_list, resize_ratios, extra_edge_padding=False,
+                      stride_ratio=1, final_size=(256, 256)):
+    """Take a series of NPZ files and combine together into single training NPZ
+
+    Args:
+        npz_list: list of NPZ files to combine
+        resize_ratios: ratio used to resize each NPZ if data is of different resolutions
```

Comment:
Is this meant to be a list or an array? does 1 indicate no scaling required?

Reply:
Yes, this is a list that matches the NPZ files being specified, where 1 indicates no resizing required

Reply:
There should be a threshold for scaling. (e.g. 1.01 does not need to re-scaled)